lib/joi-extension.js

```diff
@@ -0,0 +1,25 @@
+const regexParser = require('regex-parser')
+const regexEscape = require('escape-string-regexp')
+
+module.exports.joiStringRegexConvertible = joi => {
+  return {
+    name: 'stringRegexConvertible',
+    base: joi
+      .object()
+      .type(RegExp)
+      .required()
+      .error(() => '"search" is required and must be a regexp or a string'),
+    // eslint-disable-next-line no-unused-vars
+    coerce(value, state, options) {
+      let regex = value
+      if (typeof value !== 'string') {
+        return regex
+      } else if (value.match(/^\/.+\/[gmixXsuUAJ]*$/)) {
```

Comment:
Did you try letting `regex-parser` do all of this? It's mostly the same, though it doesn't set `'g'` as a default flag in the plain string case:

https://github.com/IonicaBizau/regex-parser.js/blob/master/test/integration/node-regex-parser-test.js

Reply:
As mentioned in #185 the problem is `my.dev.com` will match to `my-dev-com` and most likely not what you want from a string search :laughing:

Reply:
Ah, I see now! Such as shame to have to reimplement this regexp pattern, because it mostly exists inside `regexp-parser` already.

It'd be more useful if `regexParser("my.dev.com") === false` rather than turning it into `/my.dev.com/`. Ah well!

Reply:
We could say we only support `/regex/` syntax